### PR TITLE
Fix file system operations on Windows

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
When the SDK is built on Windows with _UNICODE definition defined paths were treated incorrectly.

Resolves: OLPEDGE-353

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/src/utils/Dir.cpp [hunk 1]
```diff
@@ -272,8 +294,14 @@ bool Dir::create(const std::string& path) {
     }
     dir_path += token[t];
     if (!exists(dir_path)) {
-      //TODO: FIXME: misbehaves using UNICODE
-      const TCHAR* n_path = reinterpret_cast<const TCHAR*>(dir_path.c_str());
+      // TODO: FIXME: misbehaves using UNICODE
```
Comment (sergii-vostrikov) [inline olp-cpp-sdk-core/src/utils/Dir.cpp]: still misbehaves?

Comment (mykhailo-kuchma) [reply olp-cpp-sdk-core/src/utils/Dir.cpp]: Removed

**Comment by mykhailo-kuchma** [comment]
> ASCII functions would misbehave with UTF paths

Done

**Comment by mykhailo-kuchma** [comment]
> ASCII functions would misbehave with UTF paths

Done

**Pull request merged**